packages/menu/src/Menu.tsx

```diff
@@ -5,15 +5,47 @@ import { useEventListener, useEscapeKey } from '@leafygreen-ui/hooks';
 import { css, cx } from '@leafygreen-ui/emotion';
 import { uiColors } from '@leafygreen-ui/palette';
 import { transparentize } from 'polished';
+import MenuItem, { MenuItemProps } from './MenuItem';
+import FocusableMenuItem, { FocusableMenuItemProps } from './FocusableMenuItem';
 
 const rootMenuStyle = css`
   width: 200px;
   border-radius: 3px;
   box-shadow: 0 2px 6px ${transparentize(0.8, uiColors.black)};
   background-color: ${uiColors.white};
+  list-style: none;
+  margin-block-start: 0px;
+  margin-block-end: 0px;
+  padding-inline-start: 0px;
 `;
 
-interface MenuProps extends Omit<PopoverProps, 'active' | 'spacing'> {
+function isMenuItemElement(
+  element: React.ReactNode,
+): element is React.ReactElement<MenuItemProps, typeof MenuItem> {
+  return (
+    element != null &&
```

Comment:
Let's abstract these common checks? Or rather seems like this built in [React method isValidElement](https://reactjs.org/docs/react-api.html#isvalidelement) does exactly what we need!

[Source reference which is very similar!](https://github.com/facebook/react/blob/d3622d0f977def825123f1d5f4cef19888b1eaf1/packages/react/src/ReactElement.js#L507)

Reply:
Can't use isValidElement, because we want to be more specific than just is a React Component. But can definitely abstract this and put in `lib`.

Reply:
👍